flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -135,7 +142,40 @@ public static void setTokensFor(ContainerLaunchContext amContainer, Path[] paths
 		ByteBuffer securityTokens = ByteBuffer.wrap(dob.getData(), 0, dob.getLength());
 		amContainer.setTokens(securityTokens);
 	}
-	
+
+	/**
+	 * Obtain Kerberos security token for HBase.
+	 */
+	private static void obtainTokenForHBase(Credentials credentials, Configuration conf) {
+		if (UserGroupInformation.isSecurityEnabled()) {
+			LOG.info("Attempting to obtain Kerberos security token for HBase");
+			try {
+				HBaseConfiguration.addHbaseResources(conf);
+				if (!"kerberos".equals(conf.get("hbase.security.authentication"))) {
+					LOG.info("HBase has not been configured to use Kerberos.");
+					return;
+				}
+
+				LOG.info("Connecting to HBase");
+				Connection connection = ConnectionFactory.createConnection(conf);
+
+				LOG.info("Obtaining Kerberos security token for HBase");
+				Token<AuthenticationTokenIdentifier> token = TokenUtil.obtainToken(connection);
+
+				if (token == null) {
+					LOG.error("No Kerberos security token for HBase available");
+					return;
+				}
+
+				credentials.addToken(token.getService(), token);
+				LOG.info("Added HBase Kerberos security token to credentials.");
+			} catch (IOException e) {
+				LOG.error("Caught exception while trying to obtain HBase Kerberos security token.");
+				e.printStackTrace();
```

Comment:
Yes I considered that. I figured that continuing would be 'better'. But simply rethrowing would make more sense.
I'll update the patch.